### PR TITLE
[refs] Generate idents for new clauses in MBQL lib

### LINKED ISSUE #49681: Randomly generate `:ident` for new aggregations, expressions, breakouts, and explicit joins.

### PR DESCRIPTION
**Pull request by bshepherdson**
Closes #49681.

### Description

Randomly generates NanoID `:ident`s for newly created clauses in the library.
Specifically, for expressions, aggregations, breakouts, and joins.
Adds MBQL lib unit tests for this behaviour.

Currently these idents are discarded during `lib.convert` and **don't
survive a round trip to legacy**. That will come with a later change.

### How to verify

Unit tests are passing. Click around in a query and observe these idents in the REPL.

Since they don't get converted to legacy, they don't reach the wire.

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Comment by bshepherdson** [comment]
* **#50083** <a href="https://app.graphite.dev/github/pr/metabase/metabase/50083?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/metabase/metabase/50083?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `master`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: test/metabase/lib/expression_test.cljc [hunk 1]
```diff
@@ -414,7 +416,9 @@
       (is (= "newly-named-expression"
              (lib/display-name query expr)))
       (is (not= (lib.options/uuid orig-expr)
-                (lib.options/uuid expr))))
+                (lib.options/uuid expr)))
+      (is (= (:ident (lib.options/options orig-expr))
```
Comment (appleby) [inline test/metabase/lib/expression_test.cljc:420]: nit: assuming ident are going to be read in many places, could be nice to have a `lib.options/ident` accessor, similar to `lib.options/uuid` above

File: src/metabase/util/polyfills.cljc [hunk 2]
```diff
@@ -0,0 +1,6 @@
+(ns metabase.util.polyfills
+  #?@(:cljs-test ((:require ["@peculiar/webcrypto" :as webcrypto]))))
+
+;; The browser-compatible Crypto object and the `global.crypto` instance are unavailable in NodeJS before v23.0.0.
+;; Since both developers and Github CI may be using older Node versions, we install a Polyfill and include it here.
+#?(:cljs-test (do (println "Webcrypto polyfill!") (set! js/crypto (webcrypto/Crypto.))))
```
Comment (appleby) [inline src/metabase/util/polyfills.cljc:6]: nit: put an expiration date to goad someone into removing it in the future

```suggestion
#?(:cljs-test (do (println "Webcrypto polyfill! Remove me once NodeJS v23 is sufficiently ancient.") (set! js/crypto (webcrypto/Crypto.))))
```

**Review by appleby — APPROVED** [review]

File: test/metabase/lib/expression_test.cljc (hunk 1 above)
Comment (appleby) [reply test/metabase/lib/expression_test.cljc:420]: I see now that `lib.options/ident` is already added in the [next PR in the stack](https://github.com/metabase/metabase/pull/50085).

File: src/metabase/util/polyfills.cljc (hunk 2 above)
Comment (bshepherdson) [reply src/metabase/util/polyfills.cljc:6]: Good callout. I've added a detailed `TODO` comment in the code. I was going to make the `println` more prominent, but actually I only had that in there so I could confirm it was loading properly. So I've just removed the `println`; it's likely given the Node LTS for v22 (which still needs this polyfill) extends into 2027.

**Pull request merged**